### PR TITLE
more zNPCCommon work

### PR DESCRIPTION
**Pull request by 1SDANi**

### CODE REVIEW

File: src/Game/zNPCTypeCommon.cpp [hunk 1]
```diff
@@ -504,17 +530,25 @@ void xNPCBasic::PostInit()
 // func_800F4864
 #pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "__as__10xBaseAssetFRC10xBaseAsset")
 
-// func_800F4888
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "ImmTranOn__7xPsycheFv")
+void xPsyche::ImmTranOn()
+{
+    flg_psyche = flg_psyche | 1;
```
Comment (mattbruv) [inline src/Game/zNPCTypeCommon.cpp]: `flg_psyche |= 1;`

File: src/Game/zNPCTypeCommon.cpp [hunk 2]
```diff
@@ -504,17 +530,25 @@ void xNPCBasic::PostInit()
...
+bool xPsyche::ImmTranIsOn()
+{
+    return flg_psyche & 1;
+}
 
-// func_800F48B4
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "HasGoal__7xPsycheFi")
+int32 xPsyche::HasGoal(int32 goal)
+{
+    return FindGoal(goal) != NULL;
```
Comment (mattbruv) [inline src/Game/zNPCTypeCommon.cpp:550]: `return FindGoal(goal);`

**Review by mattbruv — CHANGES_REQUESTED** [review]
Build does not 🆗

File: src/Game/zNPCTypeCommon.cpp (hunk 1 above)
Comment (1SDANi) [reply src/Game/zNPCTypeCommon.cpp]: done

File: src/Game/zNPCTypeCommon.cpp (hunk 2 above)
Comment (1SDANi) [reply src/Game/zNPCTypeCommon.cpp:550]: that is an invalid line of code, and return (int32)FindGoal(goal); does not ok

**Review by mattbruv — CHANGES_REQUESTED** [review]
Build still doesn't 🆗

**Pull request closed without merging**